dynawo/sources/Models/Modelica/Dynawo/Electrical/Controls/Protections/BaseClasses.mo

```diff
@@ -0,0 +1,213 @@
+within Dynawo.Electrical.Controls.Protections;
+
+/*
+* Copyright (c) 2022, RTE (http://www.rte-france.com)
+* See AUTHORS.txt
+* All rights reserved.
+* This Source Code Form is subject to the terms of the Mozilla Public
+* License, v. 2.0. If a copy of the MPL was not distributed with this
+* file, you can obtain one at http://mozilla.org/MPL/2.0/.
+* SPDX-License-Identifier: MPL-2.0
+*
+* This file is part of Dynawo, an hybrid C++/Modelica open source suite of simulation tools for power systems.
+*/
+
+package BaseClasses "Base models for protections"
+  extends Icons.BasesPackage;
+
+  partial model BaseDistanceProtectionLine "Base model for line distance protection"
+    import Modelica;
+    import Modelica.Constants;
+    import Modelica.Blocks.Interfaces.RealInput;
+    import Dynawo.NonElectrical.Logs.Timeline;
+    import Dynawo.NonElectrical.Logs.TimelineKeys;
+
+    parameter Integer NbZones(min = 1, max = 4) "Number of zones in the distance protection scheme";
+
+    parameter Integer LineSide(min = 1, max = 2) "Side of the line where the distance protection is located";
+
+    parameter Types.Time[NbZones] tZone "Time delay for trips in zone i (for each protected zone)";
+
+    parameter Types.PerUnit[NbZones] RPu "Resistive reach of zone i in pu (base UNom, SnRef) (for each protected zone)";
+    parameter Types.PerUnit[NbZones] XPu "Reactive reach of zone i in pu (base UNom, SnRef) (for each protected zone)";
+    parameter Integer[NbZones] TrippingZone "Positive if zone i can send a tripping signal";
+
+    parameter Types.Time CircuitBreakerTime "Time required to open the circuit breaker in s";
+
+    parameter Boolean WithBlinder "True if a load blinder is used";
+    parameter Types.Angle BlinderAnglePu "Load angle of the load blinder in rad";
+    parameter Real BlinderReachPu "Reach in the Z plane of the load blinder in pu (base UNom, SnRef)";
+
+    RealInput UMonitoredPu "Monitored voltage in pu (base UNom)";
+    RealInput PMonitoredPu "Monitored active power in pu (base SnRef)";
+    RealInput QMonitoredPu "Monitored reactive power in pu (base SnRef)";
+    Boolean Blinded "True if the apparent impedance is in the load blinder area";
```

Comment:
The coding style is the following : the input variables, one blank line, the output variables, one blank line, the other variables, thanks